### PR TITLE
protocol/tx: store refdatahashes where pointers to data entries used to go

### PR DESCRIPTION
**Pull request by bobg**

### CODE REVIEW

**Comment by bobg** [comment]
PTAL

**Comment by kr** [comment]
:eyes:

File: protocol/tx/header.go [hunk 1]
```diff
@@ -1,10 +1,12 @@
 package tx
 
+import "chain/protocol/bc"
+
 type header struct {
 	body struct {
 		Version              uint64
 		Results              []entryRef
-		Data                 entryRef
+		RefDataHash          bc.Hash
```
Comment (kr) [inline protocol/tx/header.go]: The name RefDataHash seems redundant for these fields. We can see that it's a hash right in the declaration. (The validator actually doesn't care that it's a hash of anything -- it's really just 32 arbitrary bytes.) So I would prefer RefData or even just Data.

Comment (bobg) [inline protocol/tx/header.go]: OK, done.

**Pull request merged**